pkg/registry/core/service/storage/rest_test.go

```diff
@@ -2715,152 +1787,94 @@ func TestInitNodePorts(t *testing.T) {
-							NodePort:   30054,
-						},
-						{
-							Name:       "port-udp",
-							Port:       53,
-							TargetPort: intstr.FromInt(6502),
-							Protocol:   api.ProtocolUDP,
-							NodePort:   30054,
-						},
-					},
-				},
-			},
-			expectSpecifiedNodePorts: []int{30054, 30054},
-		},
-		{
-			name: "Service has two same ports with different protocols and specifies different NodePorts",
-			service: &api.Service{
-				ObjectMeta: metav1.ObjectMeta{Name: "foo"},
-				Spec: api.ServiceSpec{
-					Selector: map[string]string{"bar": "baz"},
-					Type:     api.ServiceTypeNodePort,
-					Ports: []api.ServicePort{
-						{
-							Name:       "port-tcp",
-							Port:       53,
-							TargetPort: intstr.FromInt(6502),
-							Protocol:   api.ProtocolTCP,
-							NodePort:   30055,
-						},
-						{
-							Name:       "port-udp",
-							Port:       53,
-							TargetPort: intstr.FromInt(6502),
-							Protocol:   api.ProtocolUDP,
-							NodePort:   30056,
-						},
-					},
-				},
-			},
-			expectSpecifiedNodePorts: []int{30055, 30056},
-		},
-		{
-			name: "Service has two different ports with different protocols and specifies different NodePorts",
-			service: &api.Service{
-				ObjectMeta: metav1.ObjectMeta{Name: "foo"},
-				Spec: api.ServiceSpec{
-					Selector: map[string]string{"bar": "baz"},
-					Type:     api.ServiceTypeNodePort,
-					Ports: []api.ServicePort{
-						{
-							Name:       "port-tcp",
-							Port:       53,
-							TargetPort: intstr.FromInt(6502),
-							Protocol:   api.ProtocolTCP,
-							NodePort:   30057,
-						},
-						{
-							Name:       "port-udp",
-							Port:       54,
-							TargetPort: intstr.FromInt(6502),
-							Protocol:   api.ProtocolUDP,
-							NodePort:   30058,
-						},
-					},
-				},
-			},
-			expectSpecifiedNodePorts: []int{30057, 30058},
-		},
-		{
-			name: "Service has two same ports with different protocols but only specifies one NodePort",
-			service: &api.Service{
-				ObjectMeta: metav1.ObjectMeta{Name: "foo"},
-				Spec: api.ServiceSpec{
-					Selector: map[string]string{"bar": "baz"},
-					Type:     api.ServiceTypeNodePort,
-					Ports: []api.ServicePort{
-						{
-							Name:       "port-tcp",
-							Port:       53,
-							TargetPort: intstr.FromInt(6502),
-							Protocol:   api.ProtocolTCP,
-							NodePort:   30059,
-						},
-						{
-							Name:       "port-udp",
-							Port:       53,
-							TargetPort: intstr.FromInt(6502),
-							Protocol:   api.ProtocolUDP,
-						},
-					},
-				},
-			},
-			expectSpecifiedNodePorts: []int{30059, 30059},
-		},
-	}
+	}{{
+		name:                     "Service doesn't have specified NodePort",
+		service:                  svctest.MakeService("foo", svctest.SetTypeNodePort),
+		expectSpecifiedNodePorts: []int{},
+	}, {
+		name: "Service has one specified NodePort",
+		service: svctest.MakeService("foo", svctest.SetTypeNodePort, func(s *api.Service) {
+			s.Spec.Ports = []api.ServicePort{{
```

Comment:
we should use the helpers that we created in service testing (make.go). For all tests.